src/Migrator.sol

```diff
@@ -83,72 +72,97 @@ contract Migrator is IMigrator, Initializable, Ownable {
         swapSlippage = _amount;
     }
 
+    /// @inheritdoc IMigrator
+    function setApprovals() public onlyOwner {
+        ERC20(address(weth)).safeApprove(address(balancerVault), type(uint256).max);
+        token.safeApprove(address(balancerVault), type(uint256).max);
+        balancerPoolToken.safeApprove(address(auraPool), type(uint256).max);
+        ERC20(address(sushiLpToken)).safeApprove(address(sushiRouter), type(uint256).max);
+    }
+
     /*
-        Public functions
+        External functions
     */
 
     /// @inheritdoc IMigrator
-    function calculateSlpAmounts(uint256 _slpAmount) public view returns (uint256, uint256) {
-        uint256 slpSupply = slp.totalSupply();
-        (uint256 wethReserves, uint256 tokenReserves, ) = slp.getReserves();
+    function migrate(bool _stakeBpt, uint256 _amountTokenMin, uint256 _amountWethMin) external {
+        uint256 slpBalance = sushiLpToken.balanceOf(msg.sender);
+
+        ERC20(address(sushiLpToken)).safeTransferFrom(msg.sender, address(this), slpBalance);
 
-        // Calculate the amount of tokens and WETH the user will receive
-        uint256 amountToken = (_slpAmount * tokenReserves) / slpSupply;
-        uint256 amountWeth = (_slpAmount * wethReserves) / slpSupply;
+        _unwrapSlp(_amountTokenMin, _amountWethMin);
 
-        // Calculate the minimum amounts with unrwapSlippage tolerance
-        uint256 amountTokenMin = (amountToken * (BPS - unrwapSlippage)) / BPS;
-        uint256 amountWethMin = (amountWeth * (BPS - unrwapSlippage)) / BPS;
+        _swapWethForTokenBalancer();
 
-        return (amountTokenMin, amountWethMin);
+        _depositIntoBalancerPool();
+
+        // msg.sender receives auraBPT or BPT depending on their choice to deposit into Aura pool
+        if (_stakeBpt) {
+            _depositIntoRewardsPool();
+        } else {
+            balancerPoolToken.safeTransferFrom(address(this), msg.sender, balancerPoolToken.balanceOf(address(this)));
+        }
+
+        emit Migrated(msg.sender, slpBalance, _stakeBpt);
     }
 
-    /// @inheritdoc IMigrator
-    function calculateWethWeight(uint256 _tokenAmount) public view returns (uint256) {
-        (, int256 tokenEthPrice, , , ) = priceFeed.latestRoundData();
+    receive() external payable {
+        require(_isContract(msg.sender), "Users cannot send WETH");
```

Comment:
What contracts should be able to send WETH? This doesn't entirely prevent contracts from sending WETH because they can do so in the constructor. I think it's debatable if that entirely matters but I think for the sake of ensuring things are locked down, it is something to consider.

Reply:
yeah I think there is actually a case that can be made that weth should never be sent.